watson_developer_cloud/websocket/speech_to_text_websocket_listener.py

```diff
@@ -0,0 +1,174 @@
+from twisted.internet import ssl, reactor
+
+
+class RecognizeListener:
+    def __init__(self, audio, options, recognize_callback, url, headers):
+        self.audio = audio
+        self.options = options
+        self.callback = recognize_callback
+        self.url = url
+        self.headers = headers
+
+        factory = self.WebSocketClientFactory(self.audio, self.options,
+                                          self.callback, self.url, self.headers)
+        factory.protocol = self.WebSocketClient
+
+        if factory.isSecure:
+            contextFactory = ssl.ClientContextFactory()
+        else:
+            contextFactory = None
+        connectWS(factory, contextFactory)
+
+        reactor.run()
+
+    class WebSocketClient(WebSocketClientProtocol):
+        def __init__(self, factory, audio, options, callback):
+            self.factory = factory
+            self.audio = audio
+            self.options = options
+            self.callback = callback
+            self.isListening = False
+            self.bytes_sent = 0
+            self.ONE_KB = 1000  # in bytes
+            self.TIMEOUT_PREFIX = "No speech detected for"
+            self.CLOSE_SIGNAL = 1000
+            self.TEN_MILLISECONDS = 0.01
+            super(self.__class__, self).__init__()
+
+        def build_start_message(self, options):
+            options['action'] = 'start'
+            return options
+
+        def build_close_message(self):
+            return json.dumps({'action': 'close'})
+
+        # helper method that sends a chunk of audio if needed (as required what the specified pacing is)
+        def send_audio(self, data):
+            def send_chunk(chunk, final=False):
+                self.bytes_sent += len(chunk)
+                self.sendMessage(chunk, isBinary=True)
+                if final:
+                    self.sendMessage(b'', isBinary=True)
+
+            if (self.bytes_sent + self.ONE_KB >= len(data)):
+                if (len(data) > self.bytes_sent):
+                    send_chunk(data[self.bytes_sent:len(data)], True)
+                    return
+
+            send_chunk(data[self.bytes_sent:self.bytes_sent + self.ONE_KB])
+            self.factory.reactor.callLater(self.TEN_MILLISECONDS, self.send_audio, data=data)
+            return
+
+        def extract_transcripts(self, alternatives):
+            transcripts = []
+            for alternative in alternatives:
+                transcript = {}
+                if 'confidence' in alternative:
+                    transcript['confidence'] = alternative['confidence']
+                transcript['transcript'] = alternative['transcript']
+                transcripts.append(transcript)
+            return transcripts
+
+        def onConnect(self, response):
+            log.msg("onConnect, server connected: {}".format(response.peer))
+            self.callback.on_connected()
+
+        def onOpen(self):
+            log.msg("onOpen, Websocket connection open")
```

Comment:
why are you logging messages using the twisted log? If we want to add logging to the SDK we should use a standard python logging library instead.